### DISCUSSION TITLE
Hands only?

### DISCUSSION THREAD

**Discussion by bradcerenzia** [endink/Mediapipe4u-plugin#106] [Q&A]
I got it running tonight, really excited to play with it! Question - is there a way to record hands only, or is face required? I would like to use take recorder to record my hands turning over playing cards, preferably from the top down.

**Comment by endink** [comment]
Unsupport this now, but you can lock body joint to implement it.

**Comment by Okaz02** [comment]
> I got it running tonight, really excited to play with it! Question - is there a way to record hands only, or is face required? I would like to use take recorder to record my hands turning over playing cards, preferably from the top down.

Hello, if you're still looking, I've found a method!
This approach involves using a web browser plugin (which is already available in UE5). But how does it work? Since Mediapipe supports HTML, we can take advantage of it. It also supports Android.

I conducted an experiment and built a working example. Here's how you can do it:

First, enable the web browser plugin. it worked in UE5.2
![image](https://github.com/user-attachments/assets/19f8969d-5a00-4594-bfed-fde25bcd0110)
Next, create a widget and place a WebBrowser inside it. Make sure to check "Is Variable".
![image](https://github.com/user-attachments/assets/89c7b107-ec5f-43ca-a6b4-f1fca04103ba)
Then, create a Media Player to recognize the camera. When prompted to create a texture, check the box and press "OK".
![image](https://github.com/user-attachments/assets/d0e18ef7-bcdf-4e51-ad25-944db3dfae6d)
(it is japanese sorry)
Now, create a Material to convert the texture into a texture render. Drag the Media Player Texture you just created into the material editor and connect the RGB output to both Base Color and Emissive Color (I named this material NewMaterial).
![image](https://github.com/user-attachments/assets/473cf55e-da7b-44ab-a032-58fc80964f4d)
Next, create a Render Target Texture to render this material (I named it NewTextureRenderTarget2D).
Once that's done, call the widget from the Level Blueprint.
Go back to the widget and arrange the Blueprint like this (see image).
![image](https://github.com/user-attachments/assets/fafcd7bf-0e89-442b-9f65-944b3707f1bf)
![image](https://github.com/user-attachments/assets/2b7e4475-74bc-4e98-a8fc-045e4cebe8b2)
![image](https://github.com/user-attachments/assets/1fbf6fbd-613a-4d8d-8dca-982386f3ce45)
![image](https://github.com/user-attachments/assets/539cc284-fcb0-49d4-acfe-19d206d1d448)
![image](https://github.com/user-attachments/assets/e01cb03e-6fbb-4f99-be8c-6605c99348e4)
![image](https://github.com/user-attachments/assets/2e6df7e5-4a39-4a60-a4a8-2a50d411d3e6)
![image](https://github.com/user-attachments/assets/c8e5f527-2fca-414a-89b4-402a3a30d35a)
![image](https://github.com/user-attachments/assets/5a677728-1bee-4ea8-8007-031d759741c0)
![image](https://github.com/user-attachments/assets/6bc15cf4-cd10-4a04-8a58-9575e9f80f8f)

Load string node

[hands.txt](https://github.com/user-attachments/files/19035139/hands.txt)

![download](https://github.com/user-attachments/assets/2ac4facb-1ee6-4125-8c4d-fe5051800bf6)

The On Console Message event can be called from here (see image). Right now, this setup only plots points, but with modifications (e.g., using LK tracking), you can create a system that detects only hands.

If you want to track multiple hands, increase the maxNumHands value from 2.

This implementation also uses another plugin, which you can find here:
🔗 https://www.fab.com/ja/listings/0aadd41b-c02d-4f63-9009-bffad0070ebc
and don't forget enable the plugin.
It's free, so there shouldn’t be any issues.
Let me know if anything is unclear.
